upt_macports/upt_macports.py

```diff
@@ -278,3 +279,28 @@ def package_versions(self, name):
             sys.exit(f'The command "{cmd}" failed. '
                      'Please make sure you have MacPorts installed '
                      'and/or your PATH is set-up correctly.')
+
+    def cpan_version(self, vers):
+        vers = vers.lstrip('v')
+        index = vers.find('.')
+        other_dot = vers[index+1:].find('.')
+        if index == -1 or other_dot != -1:
+            return vers
+        ret = vers[:index]
+        fractional = vers[index+1:]
+        index = 0
+        while index < len(fractional) or index < 6:
+            sub = fractional[index:index+3]
+            if len(sub) < 3:
+                sub += '0'*(3-len(sub))
+            ret = ret + '.' + str(int(sub))
+            index += 3
+        return ret
+
+    def needs_requirement(self, req, phase, frontend):
```

Comment:
shouldn't the function that you want to override have (ideally) the same number of arguments? I was (unsuccessfully) trying to get the testing to work and run into trouble with "3 versus 4 arguments" - I am not sure if that's completely related though.... In this case, is ```frontend``` really needed, can it not just be ```if self.frontend == cpan``` (it seems to me that this is set already in the ```create_package``` function before?

What I was trying to do is to make a test where you define a requirement with a more complicated specifier and then call the ```needs_requirement``` function of ```MacPortsBackend```. You've been using ```mock``` than me, but wouldn't you be able to see then how ```upt.Backend.needs_requirement``` would get called?

Reply:
> shouldn't the function that you want to override have (ideally) the same number of arguments? I was (unsuccessfully) trying to get the testing to work and run into trouble with "3 versus 4 arguments" - I am not sure if that's completely related though.... In this case, is `frontend` really needed, can it not just be `if self.frontend == cpan` (it seems to me that this is set already in the `create_package` function before?

This should be fixed now, I was trying this out with update feature since it is easier to test with update feature.
In update feature the extra 'frontend' is required since there self.frontend is not set.

> What I was trying to do is to make a test where you define a requirement with a more complicated specifier and then call the `needs_requirement` function of `MacPortsBackend`. You've been using `mock` than me, but wouldn't you be able to see then how `upt.Backend.needs_requirement` would get called?

That's an awesome idea, tests added for the same.